### PR TITLE
Framework: Cleaning up PR test driver script

### PR DESCRIPTION
**Pull request by william76**
@trilinos/framework 

@prwolfe per our conversation, this is basically the changes from #5951 but without that one edit that would break the pr script testing stuff.  This mainly focuses on cleaning up a bit of the indentation in places that have multiline strings that break the overall flow of indentation on the script using `textwrap.dedent`

Leaving Automerge off until you get a chance to have a look.

### CODE REVIEW

File: cmake/std/PullRequestLinuxDriverTest.py [hunk 1]
```diff
@@ -435,14 +443,14 @@ def createPackageEnables(arguments):
...
-
-PR_ENABLE_BOOL(Trilinos_ENABLE_''' + enable_map[arguments.job_base_name] + ''' ON)
-''')
+                f_out.write(textwrap.dedent("""\
+                    MACRO(PR_ENABLE_BOOL  VAR_NAME  VAR_VAL)
+                        MESSAGE("-- Setting ${VAR_NAME} = ${VAR_VAL}")
+                        SET(${VAR_NAME} ${VAR_VAL} CACHE BOOL "Set in $CMAKE_PACKAGE_ENABLES_OUT")
+                    ENDMACRO()
+
+                    PR_ENABLE_BOOL(Trilinos_ENABLE_''' + enable_map[arguments.job_base_name] + ''' ON)
```
Comment (prwolfe) [inline cmake/std/PullRequestLinuxDriverTest.py]: These triple quotes must also change to """ - I  looked in the testing results and found that they are being written directly as 

PR_ENABLE_BOOL(Trilinos_ENABLE_''' + enable_map[arguments.job_base_name] + ''' ON)

Comment (william76) [reply cmake/std/PullRequestLinuxDriverTest.py]: ok. I'll get that done this morning.

**Comment by william76** [comment]
@prwolfe I updated the file to address the issue you caught.  When you have a chance can you review it again and make sure all looks good.

**Comment by prwolfe** [comment]
This looks good - I should have used a format to begin with and the error would not have happened when you changed the quoting.

Thanks!

**Comment by william76** [comment]
@prwolfe Quick reminder from our conversation earlier today -- when you get a chance can you send me any quick 5-minute insructions on running the python tests in a stand-alone manner?

**Comment by prwolfe** [comment]
@william76 - I put some stuff in the PR I just set up into this branch (see https://github.com/william76/Trilinos/pull/1 ) - it's very basic and I can expand on it if needed.

**Comment by william76** [comment]
@prwolfe 
I can't replicate your steps that you posted above.  This is with sems-python/3.5.2 loaded on a RHEL7 system.

```
wcmclen@sisu:~/dev/../source/Trilinos/cmake/std/unittests (git develop) 
0 $ ./TestPullRequestLinuxDriverMerge.py --help
ImportError: No module named site
```

**Comment by william76** [comment]
@prwolfe 
Update -- I can get past that error if I'm running it from the Trilinos source directory using this command:
`python3 cmake/std/unittests/TestPullRequestLinuxDriverTest.py`
I see this error dump:
```
0 $ python3 cmake/std/unittests/TestPullRequestLinuxDriverTest.py
..........FE.............
======================================================================
ERROR: test_verifyTargetBranch_passes_with_master_target_mm_source (__main__.Test_run)
Check to see that git is in path
----------------------------------------------------------------------
Traceback (most recent call last):
  File "cmake/std/unittests/TestPullRequestLinuxDriverTest.py", line 200, in test_verifyTargetBranch_passes_with_master_target_mm_source
    m_call.assert_called_once()
  File "/projects/sems/install/rhel7-x86_64/sems/compiler/python/3.5.2/lib/python3.5/unittest/mock.py", line 583, in __getattr__
    raise AttributeError(name)
AttributeError: assert_called_once

======================================================================
FAIL: test_verifyTargetBranch_passes_with_develop_target (__main__.Test_run)
Check to see that git is in path
----------------------------------------------------------------------
Traceback (most recent call last):
  File "cmake/std/unittests/TestPullRequestLinuxDriverTest.py", line 268, in test_verifyTargetBranch_passes_with_develop_target
    '-Dsubprojects_file=../TFW_single_configure_support_scripts/package_subproject_list.cmake'])
  File "/projects/sems/install/rhel7-x86_64/sems/compiler/python/3.5.2/lib/python3.5/unittest/mock.py", line 803, in assert_called_once_with
    return self.assert_called_with(*args, **kwargs)
  File "/projects/sems/install/rhel7-x86_64/sems/compiler/python/3.5.2/lib/python3.5/unittest/mock.py", line 792, in assert_called_with
    raise AssertionError(_error_message()) from cause
AssertionError: Expected call: check_call(['ctest', '-S', 'simple_testing.cmake', '-Dbuild_name=PR-8888-test-JenkinsBaseName-7777', '-Dskip_by_parts_submit=OFF', '-Dskip_update_step=ON', '-Ddashboard_model=Experimental', '-Ddashboard_track=testTrack', '-DPARALLEL_LEVEL=20', '-Dbuild_dir=/dev/null/workspace/pull_request_test', '-Dconfigure_script=/dev/null/workspace/Trilinos/cmake/std/dummyConfig.cmake', '-Dpackage_enables=../packageEnables.cmake', '-Dsubprojects_file=../TFW_single_configure_support_scripts/package_subproject_list.cmake'])
Actual call: check_call(['ctest', '-S', 'simple_testing.cmake', '-Dbuild_name=PR-8888-test-JenkinsBaseName-7777', '-Dskip_by_parts_submit=OFF', '-Dskip_update_step=ON', '-Ddashboard_model=Experimental', '-Ddashboard_track=testTrack', '-DPARALLEL_LEVEL=29', '-Dbuild_dir=/dev/null/workspace/pull_request_test', '-Dconfigure_script=/dev/null/workspace/Trilinos/cmake/std/dummyConfig.cmake', '-Dpackage_enables=../packageEnables.cmake', '-Dsubprojects_file=../TFW_single_configure_support_scripts/package_subproject_list.cmake'])

----------------------------------------------------------------------
Ran 25 tests in 0.154s

FAILED (failures=1, errors=1)
```
git and ctest are loaded in my environment:
```
$ where git
/projects/sems/install/rhel7-x86_64/sems/utility/git/2.10.1/bin/git
$ where ctest
/projects/sems/install/rhel7-x86_64/sems/utility/cmake/3.12.2/bin/ctest
```

**Pull request closed without merging**